hw-model/src/lib.rs

```diff
@@ -137,6 +137,36 @@ pub trait HwModel {
         }
     }
 
+    fn copy_output_until_fatal_error(
+        &mut self,
+        code: u32,
+        mut w: impl std::io::Write,
+    ) -> std::io::Result<()> {
+        loop {
+            if self.soc_ifc().cptra_fw_error_non_fatal().read() == code {
```

Comment:
I'm a bit confused why a function called "*_until_fatal_error" is actually looking at a field named error_non_fatal? Can you add a comment explaining why this is?

Reply:
Typo; fixed.